### PR TITLE
Added a shader for lit particles with SimpleLighting...

### PR DESCRIPTION
**Pull request by aleks01010101**
Cleaned up particle code.

### CODE REVIEW

File: ScriptableRenderPipeline/LightweightPipeline/LWRP/ShaderLibrary/Particles.hlsl [hunk 1]
```diff
@@ -166,45 +189,70 @@ void InitializeSurfaceData(VertexOutputLit IN, out SurfaceData surfaceData)
...
 
-#if defined(_NORMALMAP)
-    half3 normalTS = normalize(UnpackNormalScale(readTexture(_BumpMap, sampler_BumpMap, IN), _BumpScale));
+#if defined(_ALPHATEST_ON)
+    clip(result - _Cutoff + 0.0001h);
+#endif
+    return result;
+}
+
+half3 AlphaModulate(half3 albedo, half alpha)
```
Comment (phi-lira) [inline ScriptableRenderPipeline/LightweightPipeline/LWRP/ShaderLibrary/Particles.hlsl]: Seems like a good fit to be in Core.hslsl

File: ScriptableRenderPipeline/LightweightPipeline/LWRP/ShaderLibrary/Particles.hlsl [hunk 2]
```diff
@@ -166,45 +189,70 @@ void InitializeSurfaceData(VertexOutputLit IN, out SurfaceData surfaceData)
...
+#if defined(_ALPHABLEND_ON) || defined(_ALPHAPREMULTIPLY_ON) || defined(_ALPHAOVERLAY_ON)
+    half result = alpha;
 #else
-    half2 metallicGloss = half2(_Metallic, _Glossiness);
+    half result = 1.0h;
 #endif
 
-#if defined(_NORMALMAP)
-    half3 normalTS = normalize(UnpackNormalScale(readTexture(_BumpMap, sampler_BumpMap, IN), _BumpScale));
+#if defined(_ALPHATEST_ON)
```
Comment (phi-lira) [inline ScriptableRenderPipeline/LightweightPipeline/LWRP/ShaderLibrary/Particles.hlsl]: There's an alpha clip/test function in Core.hlsl that could be used here.

Comment (aleks01010101) [reply ScriptableRenderPipeline/LightweightPipeline/LWRP/ShaderLibrary/Particles.hlsl]: Yes, i'll add a parameter to it with a default set to 0.0h

File: ScriptableRenderPipeline/LightweightPipeline/LWRP/ShaderLibrary/Particles.hlsl (hunk 1 above)
Comment (aleks01010101) [reply ScriptableRenderPipeline/LightweightPipeline/LWRP/ShaderLibrary/Particles.hlsl]: This one is used only by particle shaders at the moment

**Pull request merged**